### ISSUE TITLE
Where is Location of Cache?  Cannot access source file.

### ISSUE THREAD

**Issue by onebod** [PhilippC/keepass2android#874]
Hi!
What is the location of the cached file?
I found that syncing my database isn't working as the file I'm updating is not the one I have access to.
I want to get any and ALL changes that may be saved to cached file into the file I thought I was using, which I can find, but a search doesn't find any other files.
I do find a new entry when I open the database (presumably the cached one), so it's not lost, but I can't get to the file or get it synced back so I'm using the database file and cache correctly.

Error: Could not access source file: Missing file for primary:JKFiles/Tech/JKeep P10.kdbx at /storage/emulated/0/JKFiles/Tech/JKeep P10.kdbx. Loaded file from internal cache. You can still make changes in the database and synchronize them later.

I must have broken it when I changed the folder structure.

I didn't want to "Open another database" in case I lose the updates to cached one.
I don't know what "Work on internal cache only" is for or whatwould happen if I use it.
"Synchronize database" doesn't work.  (Error:  Missing file for primary:JKFiles/Tech/JKeep P10.kdbx at /storage/emulated/0/JKFiles/Tech/JKeep P10.kdbx)

I clicked Settings > Database > Export database...
Now I feel safer to close the database.
Before opening the exported version, it looks like I can unlock that database again.
It's hard to see the filepath under the words "Unlock database"
content://com.android.externalstorage.documents/document/primary%3AJKFiles%2FTech%2FJKeep%20P10.kdbx
Not sure how to find that location and even if I show hidden files, a search for JKeep (file) or JKFiles (folder) doesn't find anything I can't already get at.

I then clicked to Change database
I can now see 2 files under "Open recent database (tap to open):"
1. content://com.android.externalstorage.documents/document/primary%3AJKFiles%2FTech%2FJKeep%20P10.kdbx
2. content://com.android.externalstorage.documents/document/primary%3AJKFiles%2FTech%2FJKeepUp%20P10.kdbx

So I probably need to also open the 2nd of those and export.
Then I'll try opening the files I can get at and thought I was editing, and sync with the exported versions of these cached files.

I also see the toggle to Show local backups.
Hopefully I don't need to open each and synchronise.
I suppose I would have to 

I see a message appeared and disappeared.  I think the original message was "Updated internal cache copy of database file" but it vanished too fast.
Now it says "Source file and cache are synchronized" when I click to open the file from its location.
Sadly the changes are still not in that file.

Ah I find "Synchronize file" must just sync the open cache file with the file in its folder.
I'll move all files to a PC and sync, then move back to the phone.

I hope this is helpful info - I know it's a bit wordy, but I'll submit it in the hope it's useful UX.
If you can see anything I should do or understand differently, please advise.

Regards,
James

(Using Huawei P10 Lite)

**Comment by PhilippC** [comment]
sorry for the late reply.
Please select settings - database - export and export the database to a location from which you can then open the database again with "Change database" - "Open database".
Make sure you always create backups.
Philipp
